src/main/webapp/scroll-container.js

```diff
@@ -0,0 +1,82 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * Creates a `ScrollContainer` responsible for hiding and displaying
+ * a clickable banner to restart automatic scrolling.
+ *
+ * <p>Automatic scrolling is when the elements in the container are scrolled
+ * to the top of the container automatically.
+ */
+export class ScrollContainer extends HTMLDivElement {
+  static #SCROLL_BANNER_CLASSES =
+      'scroll-banner sticky-top p-2 text-center text-white font-weight-bold';
+  static #SCROLL_CONTAINER_CLASSES = 'mx-5 my-3 bg-light pb-3 rounded';
+  static #AUTO_SCROLL_MESSAGE = 'Click here to continue auto-scroll';
+
+  #autoScrollIsActive;
+  #scrollBanner;
+  browserScrolled;
+
+  /** Creates a `ScrollContainer`.*/
+  constructor() {
+    super();
+    this.appendChild(this.createScrollBanner());
+    this.onscroll = this.stopAutoScroll.bind(this);
+    this.className = ScrollContainer.#SCROLL_CONTAINER_CLASSES;
+    this.#autoScrollIsActive = true;
+  }
+
+  /** Creates a banner for scrolling.*/
+  createScrollBanner() {
+    this.#scrollBanner = document.createElement('div');
+    this.#scrollBanner.innerText = ScrollContainer.#AUTO_SCROLL_MESSAGE;
+    this.#scrollBanner.className = ScrollContainer.#SCROLL_BANNER_CLASSES;
+    this.#scrollBanner.onclick = this.startAutoScroll.bind(this);
+    return this.#scrollBanner;
+  }
+
+  /** De-activates the automatic scrolling of the transcript. */
+  stopAutoScroll() {
+    if (this.browserScrolled) {
+      this.browserScrolled = false;
+      return;
+    }
+    this.#autoScrollIsActive = false;
+    this.#scrollBanner.style.visibility = 'visible';
+  }
+
+  /** Re-activates the automatic scrolling of the transcript. */
+  startAutoScroll() {
+    this.#autoScrollIsActive = true;
+    this.#scrollBanner.style.visibility = 'hidden';
```

Comment:
I think it would be nice if when you clicked the start-scrolling-again thing it would immediately scroll back to the correct element.  Would it be possible to save the most recent element passed to `scrollToTopOfContainer` (with or without `autoScrollIsActive`) and scroll to it in this function?

It feels weird pressing it when the video is paused and having nothing happen...

Reply:
hmm, the transcript scroll right away whenever the banner is pressed. Though, I had to add a call for `scrollToTopOfContainer()` in the case where I handle seeking to the current transcript line.

I'm not sure it is necessary to save the most recent element because of how often `scrollToTopOfContainer()` is called.

Reply:
The problem is when the video is paused--clicking the banner appears to do nothing until the video starts playing again

Reply:
ah gotcha, sorry I missed the last sentence. I'll do that, thanks!